doc/rtd/reference/datasources/nocloud.rst

```diff
@@ -3,22 +3,40 @@
 NoCloud
 *******
 
-The data source ``NoCloud`` allows the user to provide user data and metadata
-to the instance without running a network service (or even without having a
-network at all).
+The data source ``NoCloud`` is a flexible datasource that can be used in
+multiple different ways. With NoCloud, the user can provide user data and
+metadata to the instance without running a network service (or even without
+having a network at all). Alternatively, one may use a custom webserver to
+provide configurations.
 
-You can provide metadata and user data to a local VM boot via files on a
-`vfat`_ or `iso9660`_ filesystem. The filesystem volume label must be
-``cidata`` or ``CIDATA``.
+Configuration Methods:
+======================
 
-Alternatively, you can provide metadata via the kernel command line or SMBIOS
-"serial number" option. The data must be passed in the form of a string: ::
+Method 1: Local filesystem, labeled filesystem
+----------------------------------------------
+To provide cloud-init configurations from the local filesystem, a labeled
+`vfat`_ or `iso9660`_ filesystem containing user data and metadata may
+be used. For this method to work, the filesystem volume must be labeled
+``CIDATA``.
```

Comment:
Is that correct?

cloud-utils' cloud-localds creates both [ISO](https://github.com/canonical/cloud-utils/blob/main/bin/cloud-localds#L235) and [VFAT](https://github.com/canonical/cloud-utils/blob/main/bin/cloud-localds#L241) filesystems with label ``cidata``, not ``CIDATA``.

Reply:
Both `cidata` and `CIDATA` are supported. I'd rather document one or the other, I don't feel strongly about which one to document.

Reply:
Thanks for the review @dermotbradley!